packages/block-library/src/query/edit.js

```diff
@@ -0,0 +1,240 @@
+/**
+ * Internal dependencies
+ */
+import QueryPanel from './query-panel';
+
+/**
+ * External dependencies
+ */
+import classNames from 'classnames';
+import { debounce, isUndefined, pickBy } from 'lodash';
+
+/**
+ * WordPress dependencies
+ */
+import { __ } from '@wordpress/i18n';
+import { Component, Fragment } from '@wordpress/element';
+import {
+	BlockList,
+	BlockEditorProvider,
+	InspectorControls,
+	WritingFlow,
+} from '@wordpress/block-editor';
+import { cloneBlock, createBlock } from '@wordpress/blocks';
+import { PanelBody, Placeholder, Spinner } from '@wordpress/components';
+import { compose } from '@wordpress/compose';
+import { EntityProvider } from '@wordpress/core-data';
+import { withSelect } from '@wordpress/data';
+
+const defaultFields = [
+	'core/post-title',
+	'core/post-date',
+	'core/post-author',
+	'core/post-excerpt',
+];
+
+class Edit extends Component {
+	constructor( props ) {
+		super( props );
+		this.state = {
+			editingPost: null,
+			blocksTree: {},
+		};
+
+		this.debouncedCreateBlockTree = debounce(
+			this.createBlockTree.bind( this ),
+			1000
+		);
+	}
+
+	componentDidMount() {
+		this.createBlockTree();
+		this.updateBlocks( defaultFields.map( ( f ) => createBlock( f ) ) );
+	}
+
+	componentDidUpdate( prevProps ) {
+		const { query } = this.props;
+		if ( prevProps.query !== query ) {
+			this.createBlockTree();
+		}
+	}
+
+	createBlockTree() {
+		const { editingPost, blocksTree } = this.state;
+		const { attributes, query } = this.props;
+		const { blocks } = attributes;
+		const newBlocksTree = ( query || [] ).reduce(
+			( accumulator, post ) => ( {
+				...accumulator,
+				[ post.id ]:
+					post.id === editingPost
+						? blocksTree[ post.id ]
+						: blocks.map( ( block ) =>
+								cloneBlock( block, { post } )
+						  ),
+			} ),
+			{}
+		);
+		this.setState( { blocksTree: newBlocksTree } );
+	}
+
+	cleanBlock( block ) {
+		const { name, isValid, attributes, innerBlocks } = block;
+		return {
+			name,
+			attributes: { ...attributes, post: {} },
+			innerBlocks: innerBlocks.map( ( b ) => this.cleanBlock( b ) ),
+			isValid,
+		};
+	}
+
+	updateBlocks( blocks, postId ) {
+		const { setAttributes } = this.props;
+		const { blocksTree } = this.state;
+		const cleanBlocks = blocks.map( this.cleanBlock );
+		this.setState(
+			{
+				blocksTree: { ...( blocksTree || [] ), [ postId ]: blocks },
+				editingPost: postId,
+			},
+			() => {
+				setAttributes( { blocks: cleanBlocks } );
+				this.debouncedCreateBlockTree();
+			}
+		);
+	}
+
+	render() {
+		const {
+			attributes,
+			className,
+			postList,
+			query,
+			setAttributes,
+		} = this.props;
+
+		const { criteria } = attributes;
+
+		const { editingPost, blocksTree } = this.state;
+		const settings = {};
```

Comment:
Should we retrieve the parent settings (apply the same settings as the current editor)? select('core/block-editor').getSettings()

Reply:
Makes sense and seems fine with a cursory test.